sql/mysqld.cc

```diff
@@ -3106,11 +3117,13 @@ static bool block_until_new_connection()
     DBUG_ASSERT( ! _db_is_pushed_());
 
     // Block pthread
-    while (!abort_loop && !wake_pthread && !kill_blocked_pthreads_flag)
+    bool kbpf = false;
+    while (!(kbpf = kill_blocked_pthreads_flag.load(std::memory_order_acquire))
+      && !abort_loop && !wake_pthread)
```

Comment:
Nit - because of this command now taking two lines can you add brackets for readability?